### ISSUE TITLE
Add an example of how to use patch.

### ISSUE THREAD

**Issue by brendandburns** [kubernetes-client/javascript#450]
See https://github.com/kubernetes-client/javascript/issues/443#issuecomment-628423914

**Comment by danquah** [comment]
Just to make sure I understand this before looking into doing an example.

Lets take the example from https://github.com/kubernetes-client/javascript/issues/443#issuecomment-628423914
```typescript
 const k8sNetworkingApi = kc.makeApiClient(k8s.NetworkingV1beta1Api);
  const patchedIngress = {
    metadata: {
      annotations: {
        'nginx.ingress.kubernetes.io/cors-allow-origin': '*'
      }
    }
  }

  // The following gets rejected with a 
  // "Unsupported Media Type":
  //    the body of the request was in an unknown format - 
  //    accepted media types include: application/json-patch+json, 
  //    application/merge-patch+json, application/apply-patch+yaml
  k8sNetworkingApi.patchNamespacedIngress(ingressName, namespace, patchedIngress);

  // The fix is to do this:
  k8sNetworkingApi.patchNamespacedIngress(ingressName, namespace, patchedIngress, undefined, undefined, undefined, undefined,
    { headers: { 'content-type': 'application/merge-patch+json' } });
```

The problems boils down to this (again, as I understand it).
- The generated openapi client code does not explicitly set a Content-Type for the PATCH request
- The client code _does_ set `json: true` causing `request` (the library) to set ` Content-Type: application/json`
- the api-server subsequently rejects the PATCH as it requires another Content-Type, eg. `application/merge-patch+json`

So, in other words: the generated client code that does the `PATCH` is incompatible with the api-server out of the box, and the (for this repo) official way of fixing this is to always provide the optional `options` argument and manually set the `Content-Type` header?

**Comment by saxicek** [comment]
@danquah Exactly. That is why I asked for the example - client code does not work out of the box, optional parameter which specifies content-type must be included. In my opinion it is fine to force developers to specify this explicitly - in my case it lead me to study various merge types in Kubernetes (https://www.disasterproject.com/kubernetes-kubectl-patching/). But it should be documented.

**Comment by mauriziovitale** [comment]
@brendanburns  Same here, I am not able to patch a container image.

<img width="563" alt="Screenshot 2020-07-26 at 17 24 09" src="https://user-images.githubusercontent.com/14923305/88484245-1047aa00-cf65-11ea-8aa4-a186959e4c34.png">

It works just fine when I run it from postman:
```curl -X PATCH \
  https://domain/api/v1/namespaces/namespace/pods/process-697548b7b4-d2hgn \
  -H 'Content-Type: application/json-patch+json' \
  -d '[{"op": "replace", "path": "/spec/containers/0/image", "value":"new image"}]'
